traitsui/extras/demo.py

```diff
@@ -850,28 +908,39 @@ def _handle_file(self, filename):
                     label="Output",
                     show_label=False
                 ),
+                Item(
+                    "locals",
+                    editor=ShellEditor(share=True),
```

Comment:
What is this `share=True` for?

I hope that namespace is always refreshed when a new example is loaded, because examples are typically written with that assumption and they have lots of global mutable states.

Reply:
Share means that the namespace of the shell is bound to the `locals` trait of the example, which is reset by executing code or, obviously, when you change the active example.

The code's locals don't depend on the locals variable when run; they are initialized from a different source on each run, and because of TraitDictObject behaviour, all of these are independent copies.

Reply:
I see. Thank you!